### ISSUE TITLE
Unable to enable DHCP in AdGuard Docker container

### ISSUE THREAD

**Issue by DominoDrifter** [AdguardTeam/AdGuardHome#5310]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not ask a question

### Operating system type

Linux, Other (please mention the version in the description)

### CPU architecture

AMD64

### Installation

Docker

### Setup

On one machine

### AdGuard Home version

v0.107.21

#### What did you do?
Settings > DHCP Settings, selected the interface and populated the subnet mask, range etc. but the "Enable DHCP server" button did nothing.

#### Expected result
DHCP Server becomes enabled

#### Actual result
Nothing when clicking enable, no error or indication of success/failure but on page refresh its still disabled. When clicking "Check for DHCP servers" beside it I get the error message "In order to use DHCP server a static IP address must be set. We failed to determine if this network interface is configured using static IP address. Please set a static IP address manually."

I can then somewhat enable DHCP but not really because any subsequent setting change disables DHCP, similarly when I tried setting up adguardhomesync (after the error, for troubleshooting) every sync says "updating DHCP server config" which means its posting configuration to the AdGuard API which is not being applied, presumably because of the DHCP bug as the rest of the config syncs fine.

#### Additional information
I don't use network mode "host" as port 53 is tied up on my host so instead I use a macvlan network to give Adguard its own IP address on the network, this exact docker compose file works flawlessly (with DHCP) on my Raspberry Pi but not on my Ubuntu Server (parent interface amended).

Please relax the "check for static IP" to allow DHCP/config API to work unhindered regardless of whether a static IP is detected or not, just show a warning that it may not work.

version: "2.4"
services:

  adguard:
    container_name: adguard
    image: adguard/adguardhome:latest
    restart: always
    volumes:
      - "/home/ubuntu/docker/adguard:/opt/adguardhome/work"
      - "/home/ubuntu/docker/adguard:/opt/adguardhome/conf"
    networks:
      macvlan:
        ipv4_address: 192.168.0.2

networks:
  macvlan:
    name: macvlan
    driver: macvlan
    driver_opts:
      parent: eth0
    ipam:
      config:
        - subnet: "192.168.0.0/24"
          ip_range: "192.168.0.0/24"
          gateway: "192.168.0.1"

**Comment by merlin-tc** [comment]
I am facing the same problem. I wanted to try adguard instead of pihole.
Using a macvtap bridge (container gets it's own IP)  myself with pihole and that works fine with DHCP enabled.

I also get the error that adguard couldn't determine the IP address of the interface so maybe we could use a docker variable to explicitly tell adguard which IP to use?

The reason for using macvtap and assigning an IP to the container is the clustered pie setup I am using, so it's super simple to run the container with the same config on different pie's with a shared glusterfs storage.

**Comment by matnoublanche** [comment]
Hi,

I'm facing the same problem too.

I know that there is an environment variable on PiHole to handle that, it will be very great if we can get the same thing on AdGuard.

**Comment by JoshKoiro** [comment]
I am also curious if there is a solution to this issue.

**Comment by duhmojo** [comment]
Same here. Trying to switch to AdGuard on Unraid. I assigned a host interface, here's the docker run command:

```
docker run
  -d
  --name='AdGuard-Home'
  --net='host'
  -e TZ="America/New_York"
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="testo"
  -e HOST_CONTAINERNAME="AdGuard-Home"
  -e 'TCP_PORT_3000'='3000'
  -e 'TCP_PORT_53'='53'
  -e 'UDP_PORT_53'='53'
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.webui='http://[IP]:[PORT:3000]/'
  -l net.unraid.docker.icon='https://raw.githubusercontent.com/SiwatINC/unraid-ca-repository/master/icons/adguard.png'
  -v '/mnt/user/appdata/adguard/workingdir':'/opt/adguardhome/work':'rw'
  -v '/mnt/user/appdata/adguard/config':'/opt/adguardhome/conf':'rw' 'adguard/adguardhome'
8e1d111bdea8ecee2e4eceb4ed6428ef9b7726c2e6843d41a35c0a667231ca54
```

When I access the Adguard web setup UI here's the interfaces I see (br0 is the Unraid local network IP, presumably that's the accessible interface provided by --host):

![image](https://user-images.githubusercontent.com/1214395/219263293-efd1af90-4023-4507-9589-6ed9d73cbcdd.png)

Under settings -> DHCP:

![image](https://user-images.githubusercontent.com/1214395/219263715-554122ac-3fc5-432b-978c-0e414bb014cd.png)

Trying to "Check for DHCP servers" results in:

```
If you want to enable DHCP server anyway, make sure that there is no other active DHCP server in your network, as this may break the Internet connectivity for devices on the network!
```

But really the error toast tells a more accurate tale:

![image](https://user-images.githubusercontent.com/1214395/219264092-c111debf-aec8-4f32-a9d7-fa2bb947dc05.png)

So I found a Docker setting regarding the custom network type (macvlan and ipvlan) and the "Host access to custom networks" setting:

![image](https://user-images.githubusercontent.com/1214395/219266943-68b314bd-bf68-4432-968a-34331d69428d.png)

I tried enabling host access to custom networks and using macvlan and ipvlan and restarted the AdGuard container and still can't get the DHCP interface to work.

Can't you just add a setting the force/ignore the DHCP check? I know my network. I understand if there's a configuration where a container can't get broadcast UDP it won't work, but honestly I can't tell if the problem is some basic logic check that doesn't fit the situation.

**Comment by duhmojo** [comment]
Ok, I found that the Unraid (default?) network configuration had bridging and bonding enabled. I don't know how that was switched on because I have 1 NIC and I didn't expect to bridge everything. I assume it's a change they made.

So I disabled bonding and bridging, static IP of course:

![image](https://user-images.githubusercontent.com/1214395/219269145-d4aafb5a-6b61-48bc-abae-5758ba357656.png)

Host network looks pretty basic now:

```
docker0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.0.0  broadcast 172.17.255.255
        ether 02:42:00:e2:76:74  txqueuelen 0  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.2.5  netmask 255.255.255.0  broadcast 0.0.0.0
        ether 6c:4b:90:34:1c:c0  txqueuelen 1000  (Ethernet)
        RX packets 21531785  bytes 31088518253 (28.9 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 5835049  bytes 6218484450 (5.7 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
        device interrupt 17  memory 0xf7200000-f7220000  

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 2489  bytes 481914 (470.6 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 2489  bytes 481914 (470.6 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

Inside the container:

```
/opt/adguardhome/work # ifconfig
docker0   Link encap:Ethernet  HWaddr 02:42:00:E2:76:74  
          inet addr:172.17.0.1  Bcast:172.17.255.255  Mask:255.255.0.0
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

eth0      Link encap:Ethernet  HWaddr 6C:4B:90:34:1C:C0  
          inet addr:192.168.2.5  Bcast:0.0.0.0  Mask:255.255.255.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:21532365 errors:0 dropped:0 overruns:0 frame:0
          TX packets:5835456 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:31088581796 (28.9 GiB)  TX bytes:6218633460 (5.7 GiB)
          Interrupt:17 Memory:f7200000-f7220000 

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:2490 errors:0 dropped:0 overruns:0 frame:0
          TX packets:2490 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:481978 (470.6 KiB)  TX bytes:481978 (470.6 KiB)
```

Re-ran the AdGuard wizard and same story with DHCP setup, but I do see eth0 in there now:

![image](https://user-images.githubusercontent.com/1214395/219269632-5d92f2cb-cd89-4e21-8d5d-1d12dad94259.png)

**Comment by duhmojo** [comment]
OK! I just dropped to the console and edited the AdGuardHome.yaml file:

```
dhcp:
  enabled: true
  interface_name: eth0
  local_domain_name: lan
  dhcpv4:
    gateway_ip: 192.168.2.1
    subnet_mask: 255.255.255.0
    range_start: 192.168.2.100
    range_end: 192.168.2.200
    lease_duration: 86400
    icmp_timeout_msec: 1000
    options: []
  dhcpv6:
    range_start: ""
    lease_duration: 86400
    ra_slaac_only: false
    ra_allow_slaac: false
```

I of course could have screwed things up, but manually enabling it by passes the check. I flipped my router DHCP off, connected my phone to Wifi and I see it in the DHCP leases list.

Not satisfying but it works.

**Comment by aheath70** [comment]
I also saw this behaviour. I modified the AdGuardHome.yaml file and restarted the docker container, and it is now working. For now, at least it is working, but it would be good to get this issue resolved.

**Comment by davidpan** [comment]
Click the "Check Server" button. A red warning message will appear. Once you have seen the message, the "Enable DHCP Server" button will be enabled.

**Comment by StephaneAdeso** [comment]
I just installed adguard in docker + docker compose + portainer in my new raspberry pi 5 with fresh installation and i have the same issue. Impossible to enable dhcp server. Getting the error:

In order to use DHCP server a static IP address must be set. AdGuard Home failed to determine if this network interface is configured using a static IP address. Please set a static IP address manually.

Even if my raspberry already ha ve a static ip.

I also tried to modify the yaml, but it is read only. Any solution to this?

**Comment by StephaneAdeso** [comment]
To solve this, i had to remove the read only permition with the command:
`sudo chmod 666 /mnt/data/adguard/confdir/AdGuardHome.yaml`
 Next i went to the file  `/mnt/data/adguard/confdir/AdGuardHome.yaml` and had to change like @duhmojo say in his post. Now everything works. But this bug should be solved. Very very anoying.

**Comment by bonilha** [comment]
@duhmojo it works.

**Comment by amadorpozo** [comment]
Suffering this issue with the home-assistant addon of AdGuard (this is docker I guess). Same error, problem is that it is not obvious to access to the .yaml file to edit it... I will try to install a different DHCP server

**Comment by robert-bialy** [comment]
It took me a while but @duhmojo response saved the day. I also had to make sure that dhcpcd and systemd-networkd were disabled.

**Comment by duhmojo** [comment]
I keep getting notifications from users that managed to get this working as a container as well. I figured I'd mention that I now run Proxmox and AdGuard as an LXC container. Zero headaches. LXC containers don't have the same nightmare of network abstraction for anything mildly more complicated than basic UDP or TCP ports.

**Comment by anduz** [comment]
Hi, this issue seems to be still there.
Did anyone find another solution than editing the .yaml?

**Comment by firthm01** [comment]
I had this problem in v0.107.61 and eventually, after trying lots of suggestions, found a really simple workaround.

**Scenario**: I have a static IPv4 set up for HomeAssistant through the HA Settings->System->Network menu. Despite this, I ran in to the issue where the "Enable DHCP server" button was disabled in AdGuard and only "Check for DHCP servers" was enabled. On clicking that, I got the same warning about "a static IP must be set" and I could not enable DHCP.

**Solution**: I simply used Chrome and "Inspect Element" on the "Enable DHCP Server" button, and removed the "disabled" attribute from the button tag, and clicked it. I then set my DHCP IPv4 Settings and used "Inspect Element" again to enable the "Save Settings" button, and clicked it. All seems to be working - 2 days in so far and still serving addresses. I'll keep an eye on it.

**TLDR**; it seems the buttons are hooked up but just disabled. Re-enable them in Dev Tools. Worked for me.

**Comment by mojonba** [comment]
I am having this issue with a brand new image of Home Assistant OS and the Ad Guard Add On. Any alternative solution will be welcomed.
